src/Core/Manifest/ModuleManifest.php

```diff
@@ -225,4 +238,31 @@ public function getModules()
     {
         return $this->modules;
     }
+
+    /**
+     * Get modules sorted by priority
+     *
+     * @return Module[]
+     */
+    public function getSortedModules()
+    {
+        if (!$this->sortedModules) {
+            $order = static::config()->uninherited('module_priority');
+            $project = static::config()->get('project');
+
+            $sorter = Injector::inst()->createWithArgs(
+                PrioritySorter::class.'.modulesorter',
+                [
+                    $this->modules,
+                    $order
+                ]
+            )
+            ->setVariable(self::PROJECT_KEY, $project)
+            ->setDefaultTop(self::PROJECT_KEY);
```

Comment:
Things get really stuffed up when multiple configs specify `module_priority`, as the arrays merge, and you get something like:

```
[
  '$project',
  'blog',
  '$project',
  '$project',
  '$other_modules',
  'app',
  '$other_modules'
]
```

 I'm not sure there's a great way to solve this, short of making the priority a scalar value.